### PR TITLE
fix TextViewer postSelectionChanged for jdt ParameterNamesCodeMiningTest

### PR DESCRIPTION
**Pull request by jukzi**
https://github.com/eclipse-jdt/eclipse.jdt.ui/issues/1236

### CODE REVIEW

File: bundles/org.eclipse.jface/src/org/eclipse/jface/util/Throttler.java [hunk 1]
```diff
@@ -97,13 +97,27 @@ public Throttler(Display display, Duration minWaitTime, Runnable runnable) {
...
+
+	private void throttledExec(boolean allowSyncExec) {
 		if (display.isDisposed()) {
 			return;
 		}
 		if (scheduled.compareAndSet(false, true)) {
 			boolean exception = true;
 			try {
-				if (Thread.currentThread() == display.getThread()) {
+				if (allowSyncExec && Thread.currentThread() == display.getThread()) {
```
Comment (iloveeclipse) [inline bundles/org.eclipse.jface/src/org/eclipse/jface/util/Throttler.java:120]: Somehow the condition and the code do not match. `if(allowSyncExec) timerExec else asyncExec` makes no sense. Should it be something like `if forceTimerExec timerExec else asyncExec`? 

TimerExec is also asynchronous, argument name is misleading.

Comment (jukzi) [reply bundles/org.eclipse.jface/src/org/eclipse/jface/util/Throttler.java:120]: "allow"+"sync" vs "async"

File: bundles/org.eclipse.jface/src/org/eclipse/jface/util/Throttler.java [hunk 2]
```diff
@@ -97,13 +97,27 @@ public Throttler(Display display, Duration minWaitTime, Runnable runnable) {
 	 * returning.
 	 */
 	public void throttledExec() {
+		throttledExec(true);
+	}
+
+	/**
+	 * Like {@link #throttledExec()} but guaranteed to not run the Runnable before
+	 * this method returns.
```
Comment (iloveeclipse) [inline bundles/org.eclipse.jface/src/org/eclipse/jface/util/Throttler.java]: `guaranteed to not run the Runnable before this method returns.`
The code doesn't match the contract.
One can put breakpoint at line 124 and the code on display will run before the return, assuming current call is not in the UI thread.

Comment (jukzi) [reply bundles/org.eclipse.jface/src/org/eclipse/jface/util/Throttler.java]: valid point. i fixed the javadoc

File: bundles/org.eclipse.jface/src/org/eclipse/jface/util/Throttler.java (hunk 1 above)
Comment (iloveeclipse) [reply bundles/org.eclipse.jface/src/org/eclipse/jface/util/Throttler.java:120]: `allowSyncExec` -> `allowTimerExec` ? 
Just reading the code for the first time one could assume here is something that is supposed to run synchronous, but it is never the case.

Comment (jukzi) [reply bundles/org.eclipse.jface/src/org/eclipse/jface/util/Throttler.java:120]: I would not mind to rename anything, but actually timerExec is a Runnable that is just run() synchronous when allowSyncExec and runs runner.run() synchronous. Maybe the name "timerExec" does confuse you more

**Comment by jukzi** [comment]
The failing MultiVariablePageTest.testContextActivation is unrelated regression from 0532901782056835d47d4d13741a1492eb06815c
@iloveeclipse good to merge?

File: bundles/org.eclipse.jface/src/org/eclipse/jface/util/Throttler.java (hunk 1 above)
Comment (iloveeclipse) [reply bundles/org.eclipse.jface/src/org/eclipse/jface/util/Throttler.java:120]: `timerExec is a Runnable that is just run() synchronous`

Are we talking about `Display.timerExec()`? It is asynchronous.

Comment (jukzi) [reply bundles/org.eclipse.jface/src/org/eclipse/jface/util/Throttler.java:120]: No, about Throttler.timerExec which is run() in 121 and runs Runner in 79 without Display.timerExec() as in 85

Comment (iloveeclipse) [reply bundles/org.eclipse.jface/src/org/eclipse/jface/util/Throttler.java:120]: OK, I see.

**Comment by iloveeclipse** [comment]
I'm fine with that.

**Pull request merged**